### DISCUSSION TITLE
Need help with a beginner search problem

### DISCUSSION THREAD

**Discussion by hyunwoona** [prisma/prisma#15940] [Q&A]
Hi,
I have a `Brand` table and `Product` table, which have 1-to-many relations.
`Product` table has `brandId` column, and I am trying to do a search on `Product` name and `Brand` name.

Let's say I have the following brands - products:
```
Samsung - Galaxy S20 Smartphone
Samsung - Galaxy Tab 7 Tablet
Apple - iPhone 14 Pro Max
```
Here's what I expect in terms of query and the results:
Query: 'Samsung'
Results:
```
Samsung - Galaxy S20 Smartphone
Samsung - Galaxy Tab 7 Tablet
```
Query: 'Sams Gal Smartph'
Results:
```
Samsung - Galaxy S20 Smartphone
```

How can I implement this? Looks easy enough (every word in query should be in 'brand name' + 'product name'), but it's somehow not easy because of the fact that the brand name is in a separate table from `Product`. Also, it feels like doing a join for every row and iterating every word in query string could be expensive.

**Comment by rodolphoasb** [comment]
Have you read this [section](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#filter-conditions-and-operators)? I guess there's an example very similar to this that you want! Hope it helps!

**Reply by hyunwoona** [reply]
I have read that section. However, I couldn't find any info about how to search (or string match) on a relation field.

**Comment by cryskram** [comment]
You can probably do the following:

```
const data = await prisma.brand.findMany({
    where: {
      name: {contains: "Sams"}, 
      products: {
        some: {
          title: { contains: "Smart"}
        }
      }
    }
  });
```

hope that gets the job done!

**Reply by hyunwoona** [reply]
Thanks for your response. However, there's no way to tell that "Sams" belongs to the brand query, and "Smart" belongs to the products query.
For query: 'Sams Gal Smartph', I need the following logic in prisma:
Every word in query ("Sams", "Gal", "Smartph"), needs to be a substring of the product name or the corresponding brand name.

The following is what I can think of, and it doesn't even work when I tried it. Also, I feel like this will be awfully inefficient and I'll need something involving indexing and full-text searching.
```
const queryWords = query.split(" "); // ["Sams", "Gal", "Smartph"]
await prisma.product.findMany({
  where: {
    AND: queryWords.map((word) => {
      return {
        OR: [
          {
            name: {
              contains: word,
              mode: "insensitive",
            },
            brand: {
              name: {
                contains: word,
                mode: "insensitive",
              },
            },
          },
        ],
      };
    }),
  },
});
```

Please help me..

**Reply by cryskram** [reply]
When we write the `contains` attribute, it functions like a substring. 
For example:
```
...
   ...
      contains: "Sams"
   ...
...
```
Now this will query out all the items/instances that have the string `Sams` in them... like "Samsung", "xyzsams", or "abcsamsxzy". All these strings contain the substring __sams__. Hence the query returns them

**Reply by hyunwoona** [reply]
I understand "contains" looks for substring match. However, `query` can be multiple words ("Sams", "Gal", "Smartph" if query is "Sams Gal Smartph"), and the only way I can think of to match all the words is using `OR` and `AND` as in the example I shared.

**Reply by cryskram** [reply]
Yeah, probably using `AND` and `OR` is the way out. Maybe you could make a logic using `AND`, `OR`, `contains`, and `in` attributes

**Reply by hyunwoona** [reply]
Thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
